gateway-js/src/compose.ts

```diff
@@ -0,0 +1,40 @@
+import { resolve } from 'path';
+import { Worker } from 'worker_threads';
+import { buildSchema, GraphQLError, print } from 'graphql';
+import type { composeAndValidate, ComposedGraphQLSchema, ServiceDefinition } from '@apollo/federation';
+import type { WorkerCompositionResult } from './compose-worker';
+
+const WORKER_SCRIPT = resolve(__dirname, '..', 'dist', 'compose-worker.js');
```

Comment:
I believe this technique will pose a challenge for a bundler/tree-shaking mechanism, since it won't understand how to re-write the location of this file, or to preserve it, at runtime (files being reorganized into a single package, etc.).  I believe that using [`require.resolve`](https://nodejs.org/api/modules.html#modules_require_resolve_request_options) to refer to this in a relative-sense (i.e., `require.resolve('./compose-worker')` might be more likely to be preserved and communicate the intent to the tooling which is likely to understand this pattern?

(Unfortunately, we don't have tests for "What happens when Gateway is webpacked/rollup'd/parcel'd, etc.".)